ChurchCRM/CRM
# Public Events API Per-Event Conditional Visibility

#### What's this PR do?
Adds a field to the Events table indicating whether the event should be exposed to the Public events API.

This allows CRM event administrators to define whether a specific event is displayed on the public webpage using the WordPress CRM Calendar plugin: https://github.com/ChurchCRM/WordPress-ChurchCRM-Calendar


#### What Issues does it Close?

Closes #3032

#### Where should the reviewer start?
Note the additional field
Note the logic changes in the external/events/calendar route to filter by the new field
Note the event input / edit form changes 

#### How should this be manually tested?

1) Enable the system setting ```bEnableExternalCalendarAPI```
2)  Create an event from the calendar page with "Publicly Visible" box unchecked.
3)  Visit ```/external/calendar/events?start=2017-01-01```
4)  Confirm that the new event does not appear
5)  Edit the event, and check the publicly visible box
6)  Refresh ```/external/calendar/events?start=2017-01-01``` and confirm that the event appears

#### How should the automated tests treat this?
No.


#### Questions:
- Is there a related website / article to substantiate / explain this change?

- Does the development wiki need an update?

- Does the user documentation wiki need an update?

- Does this add new dependencies?

- Does this need to add new data to the demo database
